sno/workers/README.md

```diff
@@ -0,0 +1,215 @@
+           ]
+        },
+    "version": "3.1.0"
+    },
+    "storage": {
+        "files": [
+            {
+                "contents": {
+                    "source": "data:text/plain;charset=utf-8;base64,MTI3LjAuMC4xICAgbG9jYWxob3N0IGxvY2FsaG9zdC5sb2NhbGRvbWFpbiBsb2NhbGhvc3Q0IGxvY2FsaG9zdDQubG9jYWxkb21haW40Cjo6MSAgICAgICAgIGxvY2FsaG9zdCBsb2NhbGhvc3QubG9jYWxkb21haW4gbG9jYWxob3N0NiBsb2NhbGhvc3Q2LmxvY2FsZG9tYWluNgoxOTIuMTY4LjEyMi4yMzUgYXBpLWludC5teWNsdXN0ZXIua2FybWFsYWJzLmNvbSBhcGkubXljbHVzdGVyLmthcm1hbGFicy5jb20=",
+                    "verification": {}
+                },
+                "filesystem": "root",
+                "mode": 420,
+                "overwrite": true,
+                "path": "/etc/hosts"
+            }
+        ]
+    }
+}
+```
+
+This ignition file contains the following elements:
+
+- the source pointing to the final ignition to use
+- an overwrite of /etc/hosts so that api.$cluster.$domain and api-int.$cluster.$domain point to the ip of the SNO
+
+```
+127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
+::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
+192.168.122.235 api-int.mycluster.karmalabs.com api.mycluster.karmalabs.com
+```
+
+Provided this content got copied to /root/config.ign and we booted from rhcos live iso, we write it to disk using the following invocation
+
+```
+coreos-installer install --ignition=/root/config.ign /dev/vda
+```
+
+Note that this is a two step process, we can't embed this ignition directly in the iso, since we need to persist it.
+
+Once booted, the nodes will issue a CSR that needs to be be signed in order for them to join the cluster
+
+```
+oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' | xargs oc adm certificate approve
+ ```
+
+After a while, we should see an output similar to the following one:
+
+```
+# oc get node
+NAME                          STATUS   ROLES           AGE     VERSION
+mycluster-sno.karmalabs.com   Ready    master,worker   176m    v1.22.3+b93fd35
+mycluster-worker-0            Ready    worker          4m41s   v1.22.3+b93fd35
+mycluster-worker-1            Ready    worker          4m38s   v1.22.3+b93fd35
+```
+
+## **Testing**
+
+### Ingress controller behaviour
+
+If we check the ingress controller by using `oc get ingresscontroller default -n openshift-ingress-operator -o yaml`, we can adjust its node placement pointing to the master role, to guarantee tha the router pod will only ever run on the SNO
+
+```
+  nodePlacement:
+    nodeSelector:
+      matchLabels:
+        node-role.kubernetes.io/master: ""
+```
+
+A full [ingress yaml](99-ingress-controller.yaml) can be provided as extra manifest at install time
+
+This could be patched to run on the workers instead, but we would need to implement manually a vip mechanism along with keepalived static pods for the ingress to run properly there. We provide such assets [here](keepalived)
```

Comment:
```suggestion
This could be patched to run on the workers instead, but we would need to implement manually a virtual IP mechanism along with keepalived static pods for the ingress to run properly there. We provide such assets [here](keepalived)
```